app/subscriber/src/features/my-reports/edit/settings/template/charts/ChartViewer.tsx

```diff
@@ -39,22 +41,44 @@ let myChart: Record<string, ChartJS> = {};
 export interface IChartViewerProps extends IChartSectionProps {
   /** Chart data */
   data?: IChartData;
+  minAxisColumnWidth?: number;
 }
 
 /**
  * Displays and update the chart.
  * @param param0 Component properties.
  * @returns Component.
  */
-export const ChartViewer: React.FC<IChartViewerProps> = ({ sectionIndex, chartIndex, data }) => {
-  const { values } = useReportEditContext();
+export const ChartViewer: React.FC<IChartViewerProps> = ({
+  sectionIndex,
+  chartIndex,
+  data,
+  minAxisColumnWidth = 30,
+}) => {
+  const { values, setFieldValue } = useReportEditContext();
+  const { isShowing: showModal, toggle: toggleModal } = useModal();
 
   const canvasRef = React.useRef<HTMLCanvasElement>(null);
 
   const section = values.sections[sectionIndex];
   const chart = section.chartTemplates[chartIndex];
   const uid = `${section.id}_${sectionIndex}`;
 
+  React.useEffect(() => {
```

Comment:
Auto resize the chart based on axis values.  However, this only works if the user opens the chart section.